src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -147,28 +93,64 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
     # Check if there are any CSV files in the specified directory
     if not any(slr_data_dir.glob("*.csv")):
         raise FileNotFoundError(f"'sea_level_rise' data files not found in: '{slr_data_dir}'")
+
     # Create an empty list to store the sea level rise datasets
-    slr_nz_list = []
-    # Loop through each CSV file in the specified directory
-    for file_path in slr_data_dir.glob("*.csv"):
-        # Read the CSV file into a pandas DataFrame using pyarrow
-        slr_region = csv.read_csv(file_path).to_pandas()
-        # Extract the region name from the file name and add it as a new column in the DataFrame
-        file_name = file_path.stem
-        start_index = file_name.find('projections_') + len('projections_')
-        end_index = file_name.find('_region')
-        region_name = file_name[start_index:end_index]
-        slr_region['region'] = region_name
-        # Append the DataFrame to the list
-        slr_nz_list.append(slr_region)
-        # Log that the file has been successfully loaded
-        log.info(f"Successfully loaded the '{file_path.name}' data file.")
-    # Concatenate all the dataframes in the list and add geometry column
-    slr_nz = pd.concat(slr_nz_list, axis=0).reset_index(drop=True)
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+    slr_nz_list = [
+        pd.read_csv(file_path) for file_path in slr_data_dir.glob("*.csv")
+    ]
+
+    # Merge Site Details dataframe and Sea level projections tables with VLM
+    slr_nz_merge_vlm = pd.merge(
+        slr_nz_list[0], slr_nz_list[1], left_on='Site ID', right_on='site',
+        how='left',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_vlm['add_vlm'] = True
+
+    # Merge Site Details dataframe and Sea level projections tables with VLM
+    slr_nz_merge_novlm = pd.merge(
+        slr_nz_list[0], slr_nz_list[2], left_on='Site ID', right_on='site',
+        how='left',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_novlm['add_vlm'] = False
```

Comment:
> seems like the way that `slr_nz_list` is defined can cause the file order to be slightly different between windows and linux (maybe)

It looks like its not about windows, since it happens in linux and windows for me.

Reply:
I think it might be the way it read the column and its name 'Site ID', or the way I used the merged function. I have changed it from list number into dictionary with each column's specific name.

Reply:
> ```python
>   File "C:\Users\lpa87\dev\Digital-Twins\src\dynamic_boundary_conditions\tide\main_tide_slr.py", line 98, in main
>     sea_level_rise_data.store_slr_data_to_db(engine)
>   File "C:\Users\lpa87\dev\Digital-Twins\src\dynamic_boundary_conditions\tide\sea_level_rise_data.py", line 182, in store_slr_data_to_db
>     slr_nz = read_slr_data_from_files(slr_data_dir)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\lpa87\dev\Digital-Twins\src\dynamic_boundary_conditions\tide\sea_level_rise_data.py", line 103, in read_slr_data_from_files
>     slr_nz_merge_vlm = pd.merge(
>                        ^^^^^^^^^
>   File "C:\Users\lpa87\Anaconda3\envs\dt-188\Lib\site-packages\pandas\core\reshape\merge.py", line 110, in merge
>     op = _MergeOperation(
>          ^^^^^^^^^^^^^^^^
>   File "C:\Users\lpa87\Anaconda3\envs\dt-188\Lib\site-packages\pandas\core\reshape\merge.py", line 703, in __init__
>     ) = self._get_merge_keys()
>         ^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\lpa87\Anaconda3\envs\dt-188\Lib\site-packages\pandas\core\reshape\merge.py", line 1179, in _get_merge_keys
>     left_keys.append(left._get_label_or_level_values(lk))
>                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\lpa87\Anaconda3\envs\dt-188\Lib\site-packages\pandas\core\generic.py", line 1850, in _get_label_or_level_values
>     raise KeyError(key)
> KeyError: 'Site ID'
> ```

I'm not encountering this error myself, but it could be related to how you're using the `pandas.merge` function. Your approach differs from mine and from what's outlined in the pandas documentation. Additionally, this usage is triggering a SonarLint issue that suggests specifying the "validate" parameter for merge. You can refer to this example for guidance: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html

Reply:
I have changed it back to `df1.merge(df2)` style. Also, I used `for` loop to reduce the space, I didn't split it into another function as I think this is a small task and used only for this purpose.

Reply:
This has been addressed in a new comment below. Please refer to that.